regolith/database.py

```diff
@@ -119,8 +120,19 @@ def dump_database(db, client, rc):
 def connect(rc):
     """Context manager for ensuring that database is properly setup and torn down"""
     client = CLIENTS[rc.backend](rc)
+    client.open()
+    chained_db = {}
     for db in rc.databases:
         load_database(db, client, rc)
+        for base, coll in client.dbs[db['name']].items():
+            if base not in chained_db:
+                chained_db[base] = {}
+            for k, v in coll.items():
+                if k in chained_db[base]:
+                    chained_db[base][k].maps.append(v)
+                else:
+                    chained_db[base][k] = ChainMap(v)
```

Comment:
This is a good first cut, but it is not fully clear what happens on insertions. That is, 

1. how are insertions of new collections and documents in a database reflected in the chained_db?
2. how are insertions of new collections and documents into the chained db inserted into a concrete db?